### PR TITLE
[REVIEW] Calculating Null Mask when creating series/columns from device type objects, solving issue 135 and 183.

### PR DESCRIPTION
**Pull request by dantegd**
Modifications to calculate the null mask when creating series and/or columns from device type arrays. Solves issues 135 and 183. 

Needed workarounds since Numba cannot perform `isnan` with `int` type columns (due to unimplemented `int64` lowering support). 

IMPORTANT:: Question that needs to be answered before merging: should `to_array(self, fillna=None)` in `colunm.py` default to `fillna=pandas`. As it is right now, it ignores masked values by default, with `=pandas` it fills them with `nan`. With current behavior tests needed to be changed, but I am unsure what the better default behavior is.

### CODE REVIEW

File: pygdf/cudautils.py [hunk 1]
```diff
@@ -266,6 +267,40 @@ def compact_mask_bytes(boolbytes):
...
+
+
+@cuda.jit
+def gpu_mask_from_devary_int(ary, bits):
+    tid = cuda.grid(1)
+    base = tid * mask_bitsize
+    for i in range(base, base + mask_bitsize):
+        if i >= len(ary):
+            break
+        if not isnan(float(ary[i])):
```
Comment (kkraus14) [inline pygdf/cudautils.py]: Casting every integer type to a float seems wasteful but not sure if we have other options, @dantegd any ideas?

**Comment by kkraus14** [comment]
LGTM other than one question regarding typecasting to float.

File: pygdf/cudautils.py (hunk 1 above)
Comment (dantegd) [reply pygdf/cudautils.py]: So the definition of nan, in particular that `nan != nan` was staring me in the face, so I changed this to 

```
if ary[i] == ary[i]:
            mask_set(bits, i)
```
which removes the need to check for int or float. @scopatz @kkraus14 does this sound correct or is there a reason not to do that that I'm missing?

Comment (scopatz) [reply pygdf/cudautils.py]: I am not super familar with how the JIT is going to transform the code here, but it seems like things like the `isnan()` should only be run in the event of a float, so for other types it doesn't make sense to convert to float.  That could yield some weird errors. Also, for some data types this conversion could be pretty expensive.

**Comment by kkraus14** [comment]
LGTM, thanks @dantegd, merging!

**Pull request merged**